tools/points-service/api.go

```diff
@@ -0,0 +1,403 @@
+package main
+
+import (
+	"context"
+	"database/sql"
+	"encoding/json"
+	"log"
+	"log/slog"
+	"net/http"
+	"strconv"
+
+	"github.com/gorilla/mux"
+)
+
+type PointsAPI struct {
+	logger *slog.Logger
+	db     *sql.DB
+	ps     *PointsService
+}
+
+func NewPointsAPI(logger *slog.Logger, db *sql.DB, ps *PointsService) *PointsAPI {
+	return &PointsAPI{logger: logger, db: db, ps: ps}
+}
+
+func (p *PointsAPI) StartAPIServer(ctx context.Context, addr string) error {
+	r := mux.NewRouter()
+	r.HandleFunc("/health", p.HealthCheck).Methods("GET")
+	r.HandleFunc("/last_block", p.GetLastBlock).Methods("GET")
+	r.HandleFunc("/{receiver_type}/{receiver_address}", p.RecomputePointsForAddress).Methods("GET")
+	r.HandleFunc("/stats", p.GetTotalPointsStats).Methods("GET")
+	r.HandleFunc("/all", p.GetAllPoints).Methods("GET")
+
```

Comment:
I think we also need an API to assign points manually no? This should be authenticated API. Look at oracle. There is already an example for this. Maybe this can be done in a separate PR also.

Reply:
Yeah I don't think we should block on that for now. We'd also have access to the DB, so we can always add entries manually.